agot-bg-game-server/src/common/ingame-game-state/westeros-game-state/mustering-game-state/player-mustering-game-state/PlayerMusteringGameState.ts

```diff
@@ -278,7 +278,10 @@ export default class PlayerMusteringGameState extends GameState<ParentGameState>
                         ? this.getPotentialUnitForUpgrading(musterings, recruitmentRegion, rule.from) != null
                         : true)
                     // Check that the mustering doesn't hit unit limits
-                    .filter(rule => this.game.getAvailableUnitsOfType(this.house, rule.to) > 0)
+                    .filter(rule => {
+                        const alreadyMusteredUnits = _.flatMap(musterings.values, m => m).filter(m => m.to == rule.to).length;
```

Comment:
* You can not provide a second argument to `flatMap`, it will default to the identity function.
* We also have to consider the units that are removed by the mustering: If I don't have any footmen left, but I upgrade one to a knight, I can then muster a new footman in the same mustering resolution.

Reply:
Indeed.. Will you take over or shall I continue here?

Reply:
You can continue. You just need to compute the number of _freed_ units, that is, the `from`s in `musterings`